qcodes/instrument_drivers/Keysight/keysight_e4980a.py

```diff
@@ -0,0 +1,313 @@
+        """
+        super().__init__(name, address, terminator=terminator, **kwargs)
+
+        self._measurement_pair = MeasurementPair(
+            "CPD",
+            ("capacitance", "dissipation_factor"),
+            ("F", "")
+        )
+
+        self.add_parameter(
+            "frequency",
+            get_cmd=":FREQuency?",
+            set_cmd=":FREQuency {}",
+            get_parser=float,
+            unit="Hz",
+            vals=Numbers(20, 2E6),
+            docstring="Gets and sets the frequency for normal measurement."
+        )
+
+        self.add_parameter(
+            "current_level",
+            get_cmd=":CURRent:LEVel?",
+            set_cmd=":CURRent:LEVel {}",
+            get_parser=float,
+            unit="A",
+            vals=Numbers(0, 0.1),
+            docstring="Gets and sets the current level for measurement signal."
+        )
+
+        self.add_parameter(
+            "voltage_level",
+            get_cmd=":VOLTage:LEVel?",
+            set_cmd=":VOLTage:LEVel {}",
+            get_parser=float,
+            unit="V",
+            vals=Numbers(0, 20),
+            docstring="Gets and sets the voltage level for measurement signal."
+        )
+
+        self.add_parameter(
+            "measurement_function",
+            get_cmd=":FUNCtion:IMPedance?",
+            set_cmd=self._set_measurement
+        )
+
+        self.add_parameter(
+            "range",
+            get_cmd=":FUNCtion:IMPedance:RANGe?",
+            set_cmd=":FUNCtion:IMPedance:RANGe {}",
+            unit='Ohm',
+            vals=Enum(0.1, 1, 10, 100, 300, 1000, 3000, 10000, 30000, 100000),
+            docstring="Selects the impedance measurement range, also turns "
+                      "the auto range function OFF."
+        )
+
+        self.add_parameter(
+            "system_errors",
+            get_cmd=":SYSTem:ERRor?",
+            docstring="Returns the oldest unread error message from the event "
+                      "log and removes it from the log."
+        )
+
+        self.add_submodule(
+            "_correction",
+            Correction4980A(self, "correction")
+        )
+
+        self.connect_message()
+
+    @property
+    def correction(self):
+        return self.submodules['_correction']
+
+    @property
+    def measure_impedance(self) -> MeasurementPair:
+        return self._get_complex_impedance()
+
+    @property
+    def measurement(self) -> MeasurementPair:
+        return self._measurement()
+
+    def _get_complex_impedance(self) -> MeasurementPair:
+        """
+        Returns the impedance in the format of (R, X), where R is the
+        resistance, and X is the reactance.
+        """
+        measurement = self.ask(":FETCH:IMPedance:CORRected?")
+        r, x = [float(n) for n in measurement.split(",")]
+        measurement_pair = MeasurementPair(
+            name="RX",
+            names=("resistance", "reactance"),
+            units=("Ohm", "Ohm")
+        )
```

Comment:
shall the carefully made definitions from E4980AMeasurements be used here? :)
```suggestion
        measurement_pair = E4980AMeasurements.RX
```

Reply:
I had the same thoughts but decided not to do it because: if I do `measurement_pair = E4980AMeasurements.RX`, and take a measurement, the class `E4980AMeasurements` will be modified -- the `RX` instance will have a pair of measurement value instead of (0.0). No big deal because it won't affect anything, but I don't like the fact it's muted.

Reply:
ok, that's fine